__tests__/action.test.ts

```diff
@@ -0,0 +1,113 @@
+import {Action, FederatePortAction, App, Origin, TimeValue, Triggers, Args, Tag} from '../src/core/internal';
+
+let intendedTagDelay: TimeValue | undefined
+let intendedTagMicrostepDelay: number
+let startUpTag: Tag
+
+class ReactorWithFederatePortAction extends App {
+    a = new Action<number>(this, Origin.logical, TimeValue.nsec(10))
+    f = new FederatePortAction<number>(this, Origin.logical)
+    
+    constructor () {
+        super(TimeValue.msec(1));
+        this.addReaction(
+            new Triggers(this.startup),
+            new Args(this.schedulable(this.a)),
+            function(this, a){
+                startUpTag = this.util.getCurrentTag()
+                a.schedule(0, 0);
+            }
+        );
+
+        this.addReaction(
+            new Triggers(this.a),
+            new Args(this.schedulable(this.f)),
+            function(this, f){
+                let intendedTag: Tag | undefined
+                if (intendedTagDelay === undefined) {
+                    intendedTag = undefined
+                } else {
+                    intendedTag = startUpTag.getLaterTag(intendedTagDelay).getMicroStepsLater(intendedTagMicrostepDelay)
+                }
+                f.schedule(0, 0, intendedTag);
+            }
+        );
+    }
+
+    public setLastTagProvisional(value: boolean) {
+        this._isLastTAGProvisional = value
+    }
+}
+
+describe('Intended tag tests', function () {
+    it('Undefined inteded tag', function() {
+        intendedTagDelay = undefined
+        let app = new ReactorWithFederatePortAction()
+        expect(() => app._start()).toThrowError("FederatedPortAction must have an intended tag from RTI.")
+    })
+    
+    it('Intended tag smaller than current tag', function() {
+        intendedTagDelay = TimeValue.nsec(9)
+        intendedTagMicrostepDelay = 0
+        let app = new ReactorWithFederatePortAction()
+        app.setLastTagProvisional(false)
+        expect(() => app._start()).toThrowError("Intended tag must be greater than current tag. Intended tag")
```

Comment:
The thrown string appears to be incomplete.

Reply:
This is meant to match the prefix because the actual time value can change (because we cannot set the current Tag outside of App). Would you prefer the regular expressions?

Reply:
I wasn't aware that matching a prefix suffices to pass the test. Is that what `toThrowError` does?

Reply:
It looks like `toThrowError` performs substring matching.

https://jestjs.io/docs/expect

**Screenshot below**:
<img width="993" alt="Screen Shot 2022-08-04 at 9 05 45 PM" src="https://user-images.githubusercontent.com/2585943/182999516-84583a06-2571-43e6-8402-f387d626edb1.png">

-----------
I think this should be enough as the error message is quite specific, however, I'm open to suggestions. :)